aspnetcore/migration/webapi.md

```diff
@@ -4,47 +4,164 @@ author: ardalis
 description: Learn how to migrate a web API implementation from ASP.NET 4.x Web API to ASP.NET Core MVC.
 ms.author: scaddie
 ms.custom: mvc
-ms.date: 12/05/2019
+ms.date: 05/20/2020
 no-loc: [Blazor, "Identity", "Let's Encrypt", Razor, SignalR]
 uid: migration/webapi
 ---
 # Migrate from ASP.NET Web API to ASP.NET Core
 
 By [Scott Addie](https://twitter.com/scott_addie) and [Steve Smith](https://ardalis.com/)
 
-An ASP.NET 4.x Web API is an HTTP service that reaches a broad range of clients, including browsers and mobile devices. ASP.NET Core unifies ASP.NET 4.x's MVC and Web API app models into a simpler programming model known as ASP.NET Core MVC. This article demonstrates the steps required to migrate from ASP.NET 4.x Web API to ASP.NET Core MVC.
+An ASP.NET 4.x Web API is an HTTP service that reaches a broad range of clients, including browsers and mobile devices. ASP.NET Core unifies ASP.NET 4.x's MVC and Web API app models into a unified programming model known as ASP.NET Core MVC. This article demonstrates the steps required to migrate from ASP.NET 4.x Web API to ASP.NET Core MVC.
 
 [View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/migration/webapi/sample) ([how to download](xref:index#how-to-download-a-sample))
 
+::: moniker range=">= aspnetcore-3.0"
+
+## Prerequisites
+
+[!INCLUDE [prerequisites](../includes/net-core-prereqs-vs-3.1.md)]
+
+## Review ASP.NET 4.x Web API project
+
+As a starting point, this article uses the *ProductsApp* project created in [Getting Started with ASP.NET Web API 2](/aspnet/web-api/overview/getting-started-with-aspnet-web-api/tutorial-your-first-web-api). In that project, a basic ASP.NET 4.x Web API project is configured as follows.
+
+In *Global.asax.cs*, a call is made to `WebApiConfig.Register`:
+
+[!code-csharp[](webapi/sample/3.x/ProductsApp/Global.asax.cs?highlight=14)]
+
+The `WebApiConfig` class is found in the *App_Start* folder and has a static `Register` method:
+
+[!code-csharp[](webapi/sample/3.x/ProductsApp/App_Start/WebApiConfig.cs)]
+
+This class configures [attribute routing](/aspnet/web-api/overview/web-api-routing-and-actions/attribute-routing-in-web-api-2), although it's not actually being used in the project. It also configures the routing table, which is used by ASP.NET 4.x Web API. In this case, ASP.NET 4.x Web API expects URLs to match the format `/api/{controller}/{id}`, with `{id}` being optional.
+
+The *ProductsApp* project includes one controller. The controller inherits from `ApiController` and contains two actions:
+
+[!code-csharp[](webapi/sample/3.x/ProductsApp/Controllers/ProductsController.cs?highlight=28,33)]
+
+The `Product` model used by `ProductsController` is a basic class:
```

Comment:
Why tell them it's a basic class?

The `ProductsController` returns `Product` model objects:

Reply:
On 2nd thought, don't even say that. This tutorial should only be about the conversion. You'll need some info like how routing worked the old way and how it works now. Assume the audience is well versed in Web API .NET 4.x and just needs help on the migration.